src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64.cs

```diff
@@ -1458,31 +1308,57 @@ public static Vector64<T> GreaterThan<T>(Vector64<T> left, Vector64<T> right)
         }
 
         /// <summary>Compares two vectors to determine if all elements are greater.</summary>
+        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
         /// <param name="left">The vector to compare with <paramref name="right" />.</param>
         /// <param name="right">The vector to compare with <paramref name="left" />.</param>
-        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
         /// <returns><c>true</c> if all elements in <paramref name="left" /> were greater than the corresponding element in <paramref name="right" />.</returns>
+        /// <exception cref="NotSupportedException">The type of <paramref name="left" /> and <paramref name="right" /> (<typeparamref name="T" />) is not supported.</exception>
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool GreaterThanAll<T>(Vector64<T> left, Vector64<T> right)
-            where T : struct => GreaterThan(left, right).As<T, ulong>() == Vector64<ulong>.AllBitsSet;
+            where T : struct
+        {
+            for (int index = 0; index < Vector64<T>.Count; index++)
+            {
+                if (!Scalar<T>.GreaterThan(left.GetElementUnsafe(index), right.GetElementUnsafe(index)))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
 
         /// <summary>Compares two vectors to determine if any elements are greater.</summary>
+        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
         /// <param name="left">The vector to compare with <paramref name="right" />.</param>
         /// <param name="right">The vector to compare with <paramref name="left" />.</param>
-        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
         /// <returns><c>true</c> if any elements in <paramref name="left" /> was greater than the corresponding element in <paramref name="right" />.</returns>
+        /// <exception cref="NotSupportedException">The type of <paramref name="left" /> and <paramref name="right" /> (<typeparamref name="T" />) is not supported.</exception>
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool GreaterThanAny<T>(Vector64<T> left, Vector64<T> right)
-            where T : struct => GreaterThan(left, right).As<T, ulong>() != Vector64<ulong>.Zero;
+            where T : struct
+        {
+            for (int index = 0; index < Vector64<T>.Count; index++)
+            {
+                if (Scalar<T>.GreaterThan(left.GetElementUnsafe(index), right.GetElementUnsafe(index)))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
```

Comment:
Is this change (and others like it) for perf reasons?

Reply:
Yes. It's more efficient for the software fallback and more inline with how other impls work.